src/main/java/pl/betoncraft/betonquest/objectives/PickupObjective.java

```diff
@@ -0,0 +1,129 @@
+package pl.betoncraft.betonquest.objectives;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Listener;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.EventHandler;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.event.entity.EntityPickupItemEvent;
+
+import pl.betoncraft.betonquest.BetonQuest;
+import pl.betoncraft.betonquest.Instruction;
+import pl.betoncraft.betonquest.api.Objective;
+import pl.betoncraft.betonquest.config.Config;
+import pl.betoncraft.betonquest.Instruction.Item;
+import pl.betoncraft.betonquest.utils.PlayerConverter;
+import pl.betoncraft.betonquest.exceptions.InstructionParseException;
+
+public class PickupObjective extends Objective implements Listener {
+
+    private final Item[] pickupItems;
+
+    private final int amount;
```

Comment:
This needs to be a VariableNumber. Im currently adding this.

Reply:
We decided to leave the code as is because the current state of the Instruction API does not allow so.
An Objective with its own Objective Data can either allow variables in its own instruction string - by using the VariableNumber - OR allow for objective variable properties. This is the case because at the time the Objective is assigned to a player, it's getDefaultDataInstruction() method is invoked. But this method does not obtain any playerID from the invoking method which means that VariableNumbers cannot be resolved.
@Wolf2323 and I discussed wheather we should change the API once again. We decided not to do so because we will rewrite the Instruction in 2.0 anyways. Changing all the existing Objectives now is just unnecessary hassle.